Jenkinsfile

```diff
@@ -17,10 +17,15 @@ pipeline {
             }
           }
           steps {
-            echo "Installing Insights..."
-            sh 'pip install --user -e .[testing]'
             echo "Testing with Pytest..."
-            sh 'pytest'
+            sh """
+                virtualenv .testenv
+                source .testenv/bin/activate
+                pip install "pycparser<=2.18"
+                pip install "pyOpenSSL<=17.5.0"
```

Comment:
@Glutexo this may not be an apples to apples comparison, but when I try this on an up-to-date RHEL 6.10 vm I get this error:
```
pip install "pyOpenSSL<=17.5.0"
You are using pip version 6.0.8, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting pyOpenSSL<=17.5.0
  Using cached https://files.pythonhosted.org/packages/79/db/7c0cfe4aa8341a5fab4638952520d8db6ab85ff84505e12c00ea311c3516/pyOpenSSL-17.5.0-py2.py3-none-any.whl
Collecting six>=1.5.2 (from pyOpenSSL<=17.5.0)
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting cryptography>=2.1.4 (from pyOpenSSL<=17.5.0)
  Using cached https://files.pythonhosted.org/packages/22/21/233e38f74188db94e8451ef6385754a98f3cad9b59bedf3a8e8b14988be4/cryptography-2.3.1.tar.gz
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/tmp/pip-build-H9mTpv/cryptography/setup.py", line 28, in <module>
        "cryptography requires setuptools 18.5 or newer, please upgrade to a "
    RuntimeError: cryptography requires setuptools 18.5 or newer, please upgrade to a newer version of setuptools
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
    
      File "<string>", line 20, in <module>
    
      File "/tmp/pip-build-H9mTpv/cryptography/setup.py", line 28, in <module>
    
        "cryptography requires setuptools 18.5 or newer, please upgrade to a "
    
    RuntimeError: cryptography requires setuptools 18.5 or newer, please upgrade to a newer version of setuptools
    
    ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-H9mTpv/cryptography
```

If the goal is to make this work in our specific test environment then the above error is not as important as making this work on a stock RHEL vm.

Reply:
I intended this PR only for the Jenkins pipeline. If there is a problem with cryptography (and thus pyOpenSSL) dependencies, it should better come to #1396. The [_insights-client_](https://github.com/RedHatInsights/insights-client/) package has _setuptools_ in its RPM [dependencies](https://github.com/RedHatInsights/insights-client/blob/a931d13279158838d9527f61a357a7d0d8d92511/etc/insights-client.spec#L25), so I guess this is why the error doesn’t happen in the real world.

Reply:
@Glutexo thanks, I just wanted to confirm.

Reply:
Some things I figured out:

* When I create a new virtual environment with `virtualenv venv`, it has `setuptools<37` installed in there even though I haven’t got the `python-setuptools` package installed.
* Even after manually removing _setuptools_ from the virtual environment, both _pycparser_ and _pyOpenSSL_ install just fine.

I would like to ask you to check that a little deeper, better in the context of #1396 than by doing these manual steps. If you can confirm this, I can try to spin up a fresh RHEL 6 instance to see whether _insights-core_ installs without relying on what the _insights-client_ RPM brings along.